extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchMergeAggregator.java

```diff
@@ -58,12 +58,20 @@ public ArrayOfDoublesSketchMergeAggregator(
   @Override
   public void aggregate()
   {
-    final ArrayOfDoublesSketch update = selector.getObject();
+    final Object update = selector.getObject();
     if (update == null) {
       return;
     }
+    final ArrayOfDoublesSketch sketch;
+    if (update instanceof ArrayOfDoublesSketch) {
```

Comment:
What was this change needed for? (Asking since I'm wondering if we're missing a call to `factory.deserialize` somewhere; it's doing something similar.)

Reply:
I needed to make that code change in order to handle constructing the sketch from a base64-encoded String.
e.g. https://github.com/frankgrimes97/druid/blob/185b6f6006ecc1fb8a178f6949eb0a3f7f15ebe0/extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/tuple/sql/ArrayOfDoublesSketchSqlAggregatorTest.java#L253-L256
Is there a cleaner way to accomplish that?

Reply:
Ah, in this case we have a special function: `complex_decode_base64('type name', 'base64 bytes')`. I think the type name for this case is `arrayOfDoublesSketch`.

Reply:
Ah, I didn't know that `COMPLEX_DECODE_BASE64` function even existed! Is it documented? (I only seem to find references to it in some unit tests)

In any case, the changes to `ArrayOfDoublesSketchMergeAggregator` are no longer required after I updated the unit test to use `COMPLEX_DECODE_BASE64`:
  https://github.com/apache/druid/pull/13887/commits/2a4b4715ed8e8bf5f8b21617412437551435d95c

Reply:
It's not documented, IIRC because we are still sorting out what the best way is to expose this to end users, so we might need to change the API. It is fine for use in a unit test though. If we change the API we'll update the test too.